### DISCUSSION TITLE
Free energy calculations using deepmd model and fix adapt/fep command

### DISCUSSION THREAD

**Discussion by dkussainova** [deepmodeling/deepmd-kit#1739] [Q&A]
Hi,

I am trying to perform free energy calculations of ions in water using the deepmd model and 'fix adapt/fep N pair deepmd scale' command in lammps. However, looks like all interactions are scaled even though in fix adapt/fep function I am choosing only atom types related to ions. For example, when I try to scale interactions of ions by 0, I am getting 0 potential energy, even though I did not include atom types of water in fix adapt/fep. I tried also to include interactions of water in fix adapt/fep with a scaling factor of 1, however, this didn't help. Do you know what can be a reason for this? Any help is appreciated. 

Thanks,
Dina

**Comment by wanghan-iapcm** [comment]
DP is a multi-body potential, so it is not easy to separate and to switch-on/off a certain pair of interaction. 

What I would suggest for your case is training two DPs one including ion-water interaction and one only for water, and you may use FEP to calculate the free energy between the two DPs. 

Please note that always set the type of H and O to 0 and 1, and any ion type to >= 2. Running lammps with the water-only model on a water-ion configuration will ignore the types >= 2. 

The `exclude_type` may also work. Assume we have one ion type 2. One may set `exclude_types` to `[[0, 2], [1, 2], [2, 2]]`, and set the `atom_ener` to `[0, 0, 0]` and switch on `set_davg_zero`. @njzjz plz check if I am correct.

**Reply by dkussainova** [reply]
Thank you a lot for your reply! I am gonna try this and see how it goes.

**Reply by njzjz** [reply]
> One may set `exclude_types` to `[[0, 2], [1, 2], [2, 2]]`, and set the `atom_ener` to `[0, 0, 0]` and switch on `set_davg_zero`.

If I understand correctly, the water-only model should set the `atom_ener` to `[null, null, 0]`. Then only ion atom energy will be zero.

**Reply by wanghan-iapcm** [reply]
Thanks!

**Comment by dingye18** [comment]
You can also refer the alchemical protocol used in [OpenMM Deepmd Plugin](https://github.com/JingHuangLab/openmm_deepmd_plugin/blob/master/tests/refer/AlchemicalProtocol.pdf).
We implement the alchemical protocol within this OpenMM plugin. DP 2. version model can be integrate with this plugin directly.
